src/controls/MenuSelectTableBorderStyle.tsx

```diff
@@ -0,0 +1,193 @@
+    label: "Groove",
+    IconComponent: BorderStyleGroove,
+  },
+  {
+    value: "ridge",
+    label: "Ridge",
+    IconComponent: BorderStyleRidge,
+  },
+  {
+    value: "inset",
+    label: "Inset",
+    IconComponent: BorderStyleInset,
+  },
+  {
+    value: "outset",
+    label: "Outset",
+    IconComponent: BorderStyleOutset,
+  },
+  {
+    value: "none",
+    label: "None",
+    IconComponent: BorderStyleNone,
+  },
+];
+
+export default function MenuSelectTableBorderStyle({
+  options = DEFAULT_BORDER_STYLE_OPTIONS,
+  label,
+  className,
+}: MenuSelectTableBorderStyleProps) {
+  const editor = useRichTextEditorContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const { classes, cx } = useStyles();
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    anchorEl ? handleClose() : setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  if (!editor?.isEditable) {
+    return null;
+  }
+
+  const IconComponent =
+    options.find(
+      (option) =>
+        editor.getAttributes("tableHeader").borderStyle === option.value ||
+        editor.getAttributes("tableCell").borderStyle === option.value
+    )?.IconComponent ?? BorderStyle;
+
+  return (
+    <>
+      <MenuButton
+        tooltipLabel={label ?? "Border style"}
+        IconComponent={IconComponent}
+        onClick={handleClick}
+        selected={open}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        aria-controls={open ? "border-style-menu" : undefined}
+      />
+      <Popper
+        transition
+        open={open}
+        anchorEl={anchorEl}
+        placement="bottom-start"
+        className={cx(classes.root, className)}
+      >
+        {({ TransitionProps }) => (
+          <Fade {...TransitionProps} timeout={100}>
+            <div>
+              <ClickAwayListener
+                mouseEvent="onMouseDown"
+                touchEvent="onTouchStart"
+                onClickAway={handleClose}
+              >
+                <Paper elevation={5}>
+                  <MenuControlsContainer
+                    className={className}
+                    direction="vertical"
+                  >
+                    {options.map((option) => (
+                      <MenuButton
+                        key={option.value}
+                        tooltipLabel={option.label ?? option.value}
+                        tooltipPlacement="right"
+                        IconComponent={option.IconComponent}
+                        onClick={() =>
+                          editor
+                            .chain()
+                            .focus()
+                            .setCellAttribute("borderStyle", option.value)
+                            .updateAttributes("tableHeader", {
+                              borderStyle: option.value,
+                            })
```

Comment:
I take it `setCellAttribute` doesn't work on table header cells, so the separate `updateAttributes` call is needed? Would be worth adding a quick comment clarifying this if so